libraries/AC_Planck/AC_Planck.h

```diff
@@ -66,7 +66,17 @@ class AC_Planck {
   //Get a position, velocity, yaw command
   bool get_posvel_cmd(Location &loc, Vector3f &vel_cms, float &yaw_cd, bool &is_yaw_rate);
 
-  uint32_t mux_rates(const float rate_up, const float rate_down){return (uint32_t(rate_up) << 16) | uint32_t(rate_down);};
+  uint32_t mux_rates(float rate_up,  float rate_down){
+    if (rate_down<0)
+      rate_down*=-1;
+
+    if (rate_up<0)
+      rate_up*=-1;
+
+    uint32_t muxed_rates = ((uint32_t(rate_up) << 16) | uint32_t(rate_down));
```

Comment:
Probably worth checking the limits on the muxed rates, since you're limited to 15 bits.  Also, sorry for the late comment, but this code should really be in the `.c` file.

Reply:
Do you mean AC_PLANCK.cpp? As for checking the rates, the limit for the APM value is 10 m/s , so it won't ever get close to the 15 limit of 326 m/s (32767 cm/s)

Reply:
moved muxed_rates definition to AC_PLANCK.cpp

Reply:
added limits to rate up/down